contentgrid-appserver-query-engine-impl-jooq/src/test/java/com/contentgrid/appserver/query/engine/jooq/EventsDispatchTest.java

```diff
@@ -0,0 +1,878 @@
+
+@SpringBootTest(properties = {
+        "spring.datasource.url=jdbc:tc:postgresql:15:///",
+        "logging.level.org.jooq.tools.LoggerListener=DEBUG"
+})
+@ContextConfiguration(classes = {TestApplication.class})
+public class EventsDispatchTest {
+    public static final Scalar<Boolean> PERMIT_ALWAYS = Scalar.of(true);
+
+    @Autowired
+    private QueryEngine queryEngine;
+
+    @Autowired
+    private TableCreator tableCreator;
+
+    @MockitoBean
+    CreateEventConsumer createEventConsumer;
+
+    @MockitoBean
+    UpdateEventConsumer updateEventConsumer;
+
+    @MockitoBean
+    DeleteEventConsumer deleteEventConsumer;
+
+    @MockitoBean
+    LinkEventConsumer linkEventConsumer;
+
+    @MockitoBean
+    UnlinkEventConsumer unlinkEventConsumer;
+
+
+    @BeforeEach
+    void setup() {
+        tableCreator.createTables(APPLICATION);
+        Mockito.reset(createEventConsumer, updateEventConsumer, deleteEventConsumer,
+                linkEventConsumer, unlinkEventConsumer);
+    }
+
+    @AfterEach
+    void cleanup() {
+        tableCreator.dropTables(APPLICATION);
+    }
+
+    private static class SomeEventFailureException extends RuntimeException {}
+
+    @Test
+    void verifyCreate_happy() {
+        var created = queryEngine.create(
+                APPLICATION,
+                EntityCreateData.builder()
+                        .entityName(ENTITY_A.getName())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_A.getName())
+                                .value(123)
+                                .build())
+                        .build(),
+                PERMIT_ALWAYS,
+                createEventConsumer
+        );
+        Mockito.verify(createEventConsumer).onEntityCreate(
+                eq(APPLICATION),
+                assertArg(entityData -> {
+                    var attribute = entityData.getAttributeByName(ATTRIBUTE_A.getName());
+                    assertThat(attribute).get().isInstanceOfSatisfying(SimpleAttributeData.class, attr ->
+                            assertThat(attr.getValue()).isEqualTo(123L)
+                    );
+                })
+        );
+
+        // Transaction completed → it exists in db
+        var exists = queryEngine.findById(APPLICATION, created.getIdentity().toRequest(), PERMIT_ALWAYS);
+        assertThat(exists).isPresent();
+    }
+
+
+    @Test
+    void verifyUpdate_happy() {
+        var created = queryEngine.create(
+                APPLICATION,
+                EntityCreateData.builder()
+                        .entityName(ENTITY_A.getName())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_A.getName())
+                                .value(123)
+                                .build())
+                        .build(),
+                PERMIT_ALWAYS,
+                createEventConsumer
+        );
```

Comment:
This piece of code is repeated a lot of times in the test. Can you extract it to a separate function, so the tests become a bit less crowded and more focussed on their actual functionality?